gui/src/components/NumberView.tsx

```diff
@@ -0,0 +1,123 @@
+import { Grid, IconButton, Stack } from "@mui/material";
+import { MoreVert as MoreVertIcon } from "@mui/icons-material";
+import { useEffect, useState } from "react";
+import { formatEther, formatGwei, parseEther, parseGwei } from "viem";
+
+import { ContextMenuWithTauri } from "./ContextMenuWithTauri";
+import { Modal } from "./Modal";
+import { Datapoint } from "./Datapoint";
+
+interface Props {
+  value: bigint | string;
+  unit?: string;
+}
+
+export function NumberView({ value, unit }: Props) {
+  const [unitValues, setUnitValues] = useState({
+    wei: "",
+    gwei: "",
+    eth: "",
+    decimal: "",
+    hex: "",
+  });
```

Comment:
wei/gwei/eth are not in the same group as decimal/hex. these are different axis
e.g.: I may want to view the same gwei value in decimal or hex

also, there are many more units other than these 3: https://www.gemini.com/cryptopedia/satoshi-value-gwei-to-ether-to-wei-converter-eth-gwei#section-ethereum-denominations-ether-to-wei-gwei-to-ether-more
for each of these, you can just map it to a number of decimal places, and use `formatUnits` from viem instead of manually handling each of them